components/dialog/dialog-fullscreen.js

```diff
@@ -95,12 +160,23 @@ class DialogFullscreen extends LocalizeCoreElement(AsyncContainerMixin(DialogMix
 		this.async = false;
 		this._autoSize = false;
 		this._hasFooterContent = false;
+		this._handleResize = this._handleResize.bind(this);
```

Comment:
Instead of re-calculating every time the window resizes, as Dave B suggested elsewhere you can use `matchMedia` to test for a match but also register to be notified if the match changes. So you could use that in combination with a private property which when changed will automatically trigger a re-render.

```javascript
constructor() {
  this._handleResize = this._handleResize.bind(this);
  this._mediaQueryList = window.matchMedia('(max-width: 616px)');
  this._handleResize();
}

connectedCallback() {
  this._mediaQueryList.addEventListener('change', this._handleResize);
}

disconnectedCallback() {
  this._mediaQueryList.removeEventListener('change', this._handleResize);
}

_handleResize() {
  this._iconType = this._mediaQueryList.matches ? 'tier1:close-large-thick' : 'tier1:close-small'
}
```

Reply:
Dave L beat me too it, was just writing this up 😛

I also just wanted to point out that an important part of Lit components is making `render` a function of the element's properties. By storing the icon type in a property (like `_hasFooterContent`) instead of calculating it inside `render`, we meet that goal and Lit will only call `render` when that value actually changes.  This limits the amount or re-renders and it will handle it automatically for us (no need for us to call `this.requestUpdate();`).

From my experience, this is an easy thing to know and understand and then accidentally overlook when deep in the weeds of a problem, so it's good to keep checking for this along the way (in your own stuff, in pull requests, etc)

Reply:
Aha, so that's how you do it! Thanks all. I've made those changes now 😄

Reply:
This API is certainly the preferred one, but I would just test the different browsers to make sure that works as expected.  Safari might have been late to the party supporting `addEventListener` and quite a whole ago when I was first trying to use those for handling color scheme, I had some trouble with it.  I'm not sure if we have anything running previous versions of Safari though.  You _might_ need to include the old method for backwards compatibility... https://developer.mozilla.org/en-US/docs/Web/API/MediaQueryList#methods

Reply:
Good memory Dave.

> Prior to Safari 14, MediaQueryList is based on EventTarget, so you must use addListener() and removeListener() to observe media query lists.

IMO as long as it works on first-render in Safari < 14 (which I think it should based on how this code works), I'd be satisfied with that and just stick with the better API.